extsort.go

```diff
@@ -131,24 +136,34 @@ func (i *Iterator) next() bool {
 		return false
 	}
 
-	section, data := i.heap.PopData()
+	section, ent := i.heap.PopEntry()
 	if err := i.fillHeap(section); err != nil {
-		bufferPool.Put(data)
+		ent.Release()
 		i.err = err
 		return false
 	}
 
-	prev := i.data
-	i.data = data
+	prev := i.ent
+	i.ent = ent
 	if prev != nil {
-		bufferPool.Put(prev)
+		prev.Release()
 	}
 	return true
 }
 
-// Data returns the data at the current cursor position.
+// Key returns the key at the current cursor position.
+func (i *Iterator) Key() []byte {
+	return i.ent.Key()
+}
+
+// Value returns the value at the current cursor position.
+func (i *Iterator) Value() []byte {
+	return i.ent.Val()
+}
+
+// Data is an alias for Key, retained for compatibility.
```

Comment:
Nit: I'd not mention anything about "compatibility" here as it kinda implies that this method should not be used in modern code. But this is more about indicating intent - if sorter is used for plain values (`Append([]byte)`) it makes sense to refer to these plain values as `Data()`.